blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScmSaveFileRequest.java

```diff
@@ -66,6 +69,14 @@ public Object save(@Nonnull String apiUrl, @Nullable String owner, @Nullable Str
         }
 
         try {
+            // First validate that you have access to push to the repository
+            // Otherwise the call to post content will fail with a 404
+            GitHub gitHub = GitHub.connectToEnterprise(apiUrl, accessToken);
```

Comment:
We should do this check in [getContent()](https://github.com/jenkinsci-cert/blueocean-plugin/blob/c03b804b6ad81158c5908885750a7ea57b3621ca/blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScmContentProvider.java#L40) as well. Better to fail fast than let user do all work then fail.